runtests.jl

```diff
@@ -50,6 +50,10 @@ function test_lang(day_num_str, input_path, reference_output, lang)
     all_sources = readdir(LANG_DIR)
     filter!(startswith(day_num_str), all_sources)
 
+    if (lang == "cpp")
+        run(`bazel build //...`)
```

Comment:
second part makes sense, why it's called once every day?

Reply:
https://github.com/Moelf/AoC2023/blob/a1999db5a3496a0b4f07d3e4dc2796a1ba7e8fd2/runtests.jl#L77

actually it's more than once a day, `test_lang` is called in the inner-most loop, so every time cpp is run in the inner-most loop it will run build?

Reply:
Yeah, but `bazel build` is smart enough to not rebuild if nothing has been changed. Since it's possible to invoke `bazel-bin/src/cpp`, I think it's reasonable to assume the built executables aren't touched while the test are being run.

Reply:
Minor overheads, but yeah I'l move it out

Reply:
yeah I can see the overhead is smaller than Julia haha.

But leaving it in there is equivalent to calling `bazel build` in the:
```julia
run_solution(source_path, input_path, ::Val{:cpp})
```

I think